org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/HoverInfoProvider.java

```diff
@@ -43,17 +47,23 @@
 
 	private static final long LOCAL_VARIABLE_FLAGS= LABEL_FLAGS & ~JavaElementLabels.F_FULLY_QUALIFIED | JavaElementLabels.F_POST_QUALIFIED;
 
+	private static final long COMMON_SIGNATURE_FLAGS = LABEL_FLAGS & ~JavaElementLabels.ALL_FULLY_QUALIFIED | JavaElementLabels.M_FULLY_QUALIFIED;
```

Comment:
so COMMON_SIGNATURE_FLAGS & JavaElementLabels.T_FULLY_QUALIFIED == 0, which means the type doesn't get FQCN'ed in JavaElementLabelComposer.appendTypeLabel